vmdb/spec/requests/api/policies_assignment_spec.rb

```diff
@@ -55,707 +57,649 @@ def app
-      expect(result["success"]).to be_true
-      expect(result["href"]).to match(object_url)
-      expect(result["#{sc_prefix}_href"]).to match("/api/#{subcollection}/#{policy.id}")
+    results_hash = policies.collect do |policy|
+      {"success" => true, "href" => object_url, "#{sc_prefix}_href" => %r{/api/#{subcollection}/#{policy.id}}}
     end
+    expect_results_to_match_hash("results", results_hash)
     expect(object.get_policies.size).to eq(policies.size)
     expect(object.get_policies.collect(&:guid)).to match_array(policies.collect(&:guid))
   end
 
   def test_policy_unassign_no_role(object_policies_url)
-    basic_authorize @cfme[:user], @cfme[:password]
+    api_basic_authorize
 
-    @success = run_post(object_policies_url, gen_request(:unassign))
+    run_post(object_policies_url, gen_request(:unassign))
 
-    expect(@success).to be_false
-    expect(@code).to eq(403)
+    expect_request_forbidden
   end
 
   def test_policy_unassign_invalid_policy(object_policies_url, collection, subcollection)
-    update_user_role(@role, subcollection_action_identifier(collection, subcollection, :unassign))
-    basic_authorize @cfme[:user], @cfme[:password]
+    api_basic_authorize subcollection_action_identifier(collection, subcollection, :unassign)
 
-    @success = run_post(object_policies_url, gen_request(:unassign, :href => "/api/#{subcollection}/999999"))
+    run_post(object_policies_url, gen_request(:unassign, :href => "/api/#{subcollection}/999999"))
 
-    expect(@success).to be_false
-    expect(@code).to eq(404)
+    expect_resource_not_found
   end
 
   def test_policy_unassign_invalid_policy_guid(object_url, object_policies_url, collection, subcollection)
-    update_user_role(@role, subcollection_action_identifier(collection, subcollection, :unassign))
-    basic_authorize @cfme[:user], @cfme[:password]
-
-    @success = run_post(object_policies_url, gen_request(:unassign, :guid => "xyzzy"))
-
-    expect(@success).to be_true
-    expect(@code).to eq(200)
-    expect(@result).to have_key("results")
-    result = @result["results"].first
-    expect(result["success"]).to be_false
-    expect(result["href"]).to match(object_url)
-    expect(result["message"]).to match(/must specify a valid/i)
+    api_basic_authorize subcollection_action_identifier(collection, subcollection, :unassign)
+
+    run_post(object_policies_url, gen_request(:unassign, :guid => "xyzzy"))
+
+    expect_request_success
+    results_hash = [{"success" => false, "href" => object_url, "message" => /must specify a valid/i}]
+    expect_results_to_match_hash("results", results_hash)
   end
 
   def test_unassign_multiple_policies(object_policies_url, collection, subcollection, options = {})
-    update_user_role(@role, subcollection_action_identifier(collection, subcollection, :unassign))
-    basic_authorize @cfme[:user], @cfme[:password]
+    api_basic_authorize subcollection_action_identifier(collection, subcollection, :unassign)
```

Comment:
Does it make sense to merge `test_unassign_multiple_policies` and `test_unassign_multiple_policy_profiles` ?